pkgs/top-level/python-packages.nix

```diff
@@ -7802,6 +7802,49 @@ let
     };
   };
 
+  # Special check phase for numpy and scipy, following the same set of steps:
+  # First "install" the package, then import what was installed, and call the
+  # .test() function, which will run the test suite.
+  numpyScipyCheckPhase = python: pkgName: ''
```

Comment:
The name `self.python` is an attribute to this package set; why do you need to take that as a function argument here?